### ISSUE TITLE
RFC 8255: Multiple Language Content Type (multipart/multilingual)

### ISSUE THREAD

**Issue by pstch** [stalwartlabs/mail-parser#35]
Hi ! First of all, thanks for your work on this library, it looks really well-made and well-maintained.

I'm trying to build a multilingual MLM, and I will be using mail-parser and mail-builder in this project. I'm wondering if you've thought about including support for RFC8255. I'd be happy to sponsor this feature, or even try to write an implementation myself, however I'm bad at API design, and wanted to ask if you have ideas about the proper API for this, especially as how to abide by Postel's law, as RFC8255 is not implemented a lot (also, I couldn't find it mentioned in JMAP specs), and for example I'm not sure how it's supposed to compose with multipart/alternative (is it `multipart/alternative >> multipart/multilingual` ? `multipart/multilingual >> multipart/alternative` ?), as well as with attachments (are multilingual attachments a thing ?). It also changes a few things in the message structure, for example RFC8255 allows to expose multiple translations of the `From` field, which could even have different mail addresses (not only names).

**tl;dr** Is this something you'd like to see implemented in `mail-parser` ? Do you have any thoughts about a proper API design for this ? Do you think it would involve a lot of changes to the existing code ?

ps: I know that this RFC is not implemented a lot, but I personally think it's unfortunate : multilingual communication tools are often overlooked, but the usage still exists, I participate in quite a few lists where I always have to scroll to get to the right translation of a message for me, I cannot filter on messages that were translated to a language I understand. Implementation has to start somewhere, and I believe that parsing/building libraries are the right place for that.

**Comment by mdecimus** [comment]
Hi, I don't think it is necessary to modify mail-parser in order to work with ``multipart/multilingual`` messages. I took a quick look at RFC8255 and the messages for each language are available as ``message/rfc822`` parts. So what you can do is iterate each part of the message until you find the message in the language you'd like to display to the user.

Out of curiosity, are you aware of any mail clients that support RFC8255?

**Comment by pstch** [comment]
Oh that's right, I can just look at message parts, and probably reuse Message to parse the translations, which can be either `message/rfc822` or `message/global`, but given that mail-parser conforms to RFC6532 (internationalized headers), it should work properly. 

The only client I'm aware of that supports RFC8255 is NeoMutt, a [pull request](https://github.com/neomutt/neomutt/pull/1104) was merged there 5 years ago. I would of course make RFC8255 support an option in the MLM, so that it can be enabled for clients that support it. Hopefully we will see more implementations of this RFC, and of email internationalization in years to come.
